### PR TITLE
Add CalcLength

### PR DESCRIPTION
**Pull request by nattridgell**
Adds:
- CalcLength constructor
- A couple of tests for CalcLength
- fromValue / fromDictionary in LengthValue calls
  corresponding constructors for SimpleLength / CalcLength
- extremely simple tests for fromValue / fromDictionary
- remove unnecessary tags for consistency
  (@constructor, @enum tags)

### CODE REVIEW

File: src/calc-length.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      throw(new TypeError('CalcLength must be passed a dictionary object'));
+    }
+
+    // NOTE: Does not check dictionary for invalid fields.
+    for (var type in LengthValue.LengthType) {
+      var value = dictionary[type];
+      if (typeof value == 'number') {
+        this[type] = value;
+      } else if (typeof value == 'string') {
+        var nValue = Number.parseFloat(value);
```
Comment (wilddamon) [inline src/calc-length.js]: Need to check with Shane if we care about parsing strings here

File: test/js/length-value.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+suite('LengthValue', function() {
+  test('fromValue returns instanceof a LengthValue and StyleValue', function() {
```
Comment (wilddamon) [inline test/js/length-value.js]: Should these also check for SimpleLength and CalcLength types here?

File: src/calc-length.js (hunk 1 above)
Comment (nattridgell) [inline src/calc-length.js]: I'll remove it for now and then we can add it back in later if need be.

**Comment by nattridgell** [comment]
I hope this is what you meant :-)

File: src/calc-length.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+   * TODO: CalcLength(cssString)
+   */
+  function CalcLength(dictionary) {
+    if (typeof dictionary != 'object') {
+      throw new TypeError('CalcLength must be passed a CalcDictionary object.');
+    }
+
+    var hasValidLength = false;
+
+    // NOTE: Does not check dictionary for invalid fields.
```
Comment (wilddamon) [inline src/calc-length.js]: Please add a comment saying the check is done in fromDictionary.

File: src/length-value.js [hunk 4]
```diff
@@ -65,11 +64,44 @@
...
+        firstValidType = type;
+      } else if (value != null && value != undefined) {
+        throw new TypeError('Value of each length must be null or a number.');
+      }
+    }
+
+    if (numLengthsFound < 1) {
+      throw new TypeError('A CalcDictionary must have at least one valid length.');
+    } else if (numLengthsFound < 2) {
+      return new SimpleLength(dictionary[firstValidType], firstValidType);
```
Comment (wilddamon) [inline src/length-value.js]: As discussed, don't do this for consistency with calc length constructor - a calc with a single value is ok.

File: src/length-value.js [hunk 5]
```diff
@@ -65,11 +64,44 @@
...
+    } else {
+      return new CalcLength(dictionary);
```
Comment (wilddamon) [inline src/length-value.js]: As discussed, we don't want to do this any more - you can move all of this logic into the constructor, and get rid of the part where you create a SimpleLength.
Sorry about that...

**Comment by wilddamon** [comment]
LGTM

**Pull request merged**